src/backend/api/Fusion.Resources.Domain/Queries/Departments/GetRelevantDeparmentProfile.cs

```diff
@@ -0,0 +1,322 @@
+{
+    /// <summary>
+    /// Fetch the profile for a resource owner. Will compile a list of departments the person has responsibilities in and which is relevant.
+    /// </summary>
+    public class GetRelevantDeparmentProfile : IRequest<IEnumerable<QueryRelevantDepartmentProfile>>
+    {
+        public GetRelevantDeparmentProfile(string profileId, AspNetCore.OData.ODataQueryParams query)
+        {
+            ProfileId = profileId;
+            Query = query;
+        }
+
+        /// <summary>
+        /// Mail or azure unique id
+        /// </summary>
+        public PersonId ProfileId { get; set; }
+        public ODataQueryParams Query { get; }
+
+
+
+        public class Handler : IRequestHandler<GetRelevantDeparmentProfile, IEnumerable<QueryRelevantDepartmentProfile>>
+        {
+            private readonly TimeSpan defaultAbsoluteCacheExpirationHours = TimeSpan.FromHours(1);
+            private readonly ILogger<Handler> logger;
+            private readonly IFusionProfileResolver profileResolver;
+            private readonly IFusionRolesClient rolesClient;
+            private readonly IMediator mediator;
+            private readonly ILineOrgResolver lineOrgResolver;
+            private readonly IMemoryCache memCache;
+            private bool sapIdFound = false;
+            List<QueryRelevantDepartmentProfile?> lstDepartments = new List<QueryRelevantDepartmentProfile?>();
+
+            public Handler(ILogger<Handler> logger, IFusionProfileResolver profileResolver, IFusionRolesClient rolesClient, ILineOrgResolver lineOrgResolver, IMediator mediator, IMemoryCache memCache)
+            {
+                this.logger = logger;
+                this.profileResolver = profileResolver;
+                this.rolesClient = rolesClient;
+                this.mediator = mediator;
+                this.lineOrgResolver = lineOrgResolver;
+                this.memCache = memCache;
+
+            }
+            private CancellationToken _cancellationToken;
+            public async Task<IEnumerable<QueryRelevantDepartmentProfile>> Handle(GetRelevantDeparmentProfile request, CancellationToken cancellationToken)
+            {
+                _cancellationToken = cancellationToken;
+                var user = await profileResolver.ResolvePersonBasicProfileAsync(request.ProfileId.OriginalIdentifier);
+
+                if (user is null) return null;
+
+                // Resolve departments with responsibility
+                var departmentsWithAccess = await ResolveDepartmentsWithAccessAsync(user);
+
+                if (user.IsResourceOwner)
+                {
+                    List<string> Reasons = new List<string>();
+                    Reasons.Add("Manager");
+
+                    if (user.FullDepartment != null)
+                    {
+
+                        QueryRelevantDepartmentProfile? DepartmentProfile = await GetDepartmentInformation(user.FullDepartment, Reasons, request.Query);
```

Comment:
variables should not be PascalCased.

async methods should have Async suffix. => await GetWhateverAsync()